### PR TITLE
Add method accessible_from to test security group rules

### PR DESCRIPTION
**Pull request by temyers**
Added `accessible_from?` to security groups and RDS instances.

The semantics for security groups are:
 * There is an ingress rule where the CIDR matches, or is a superset of the supplied CIDR range.

The semantics for RDS instances is slightly different:
 * Is there a security group which satisfies the CIDR
 * If the CIDR is for a public subnet, is the database publicly accessible

There are additional semantics that I consider out of scope, since they should probably be tested separately:
 * If the CIDR is an internal address, is it within the CIDR of the associated VPC (if the RDS instance is in a VPC)
 * VPC Peering CIDRs

### CODE REVIEW

File: lib/resources/rds/db_instance.rb [hunk 1]
```diff
@@ -54,6 +56,21 @@ def with_encrypted_storage?
             @dbi.storage_encrypted
           end
 
+          # Tests whether the database should allow connections from the given CIDR range.
```
Comment (temyers) [inline lib/resources/rds/db_instance.rb]: Is the documentation sufficient?

File: lib/resources/rds/db_instance.rb [hunk 2]
```diff
@@ -54,6 +56,21 @@ def with_encrypted_storage?
...
+          # Inspects each of the security groups associated with the instance and associated inbound rules
+          # to check for a rule matching the given CIDR range.
+          # For external connections, it is verified that the database is publicly accessible.
+          def accessible_from?(cidr)
```
Comment (temyers) [inline lib/resources/rds/db_instance.rb]: Given the description and caveats, is this an appropriate name?

It's effectively testing the security rules do not prevent connection - not that you would actually be able to connect.

File: lib/resources/rds/db_instance.rb [hunk 3]
```diff
@@ -54,6 +56,21 @@ def with_encrypted_storage?
...
+
+
+            vpc_security_groups().each do |sg|
+              return true if AWS::EC2::SecurityGroup.new(sg, @ec2).accessible_from?(cidr)
+            end
+
```
Comment (temyers) [inline lib/resources/rds/db_instance.rb]: TODO - implement check on public accessibility.

File: spec/ec2/security_group_spec.rb [hunk 4]
```diff
@@ -1,13 +1,52 @@
...
-ec2.stub_responses(
-  :describe_security_groups,
-  security_groups: [
-    {
-      owner_id: '000000000000000000000',
-      group_name: 'test-group',
-      description: 'test-group description',
-      ip_permissions: [
+
+def security_group
```
Comment (temyers) [inline spec/ec2/security_group_spec.rb]: Not really happy with this implementation - is there a better option?

File: spec/ec2/security_group_spec.rb [hunk 5]
```diff
@@ -1,13 +1,52 @@
...
+    @security_group[attrib] = value
+    @security_group
+  end
+end
+
+RSpec.context "Security Group allows ingress from anywhere" do
+  ec2.stub_responses(
+    :describe_security_groups,
+    security_groups: [
+      security_group.with(:ip_permissions, [
```
Comment (temyers) [inline spec/ec2/security_group_spec.rb]: the fixture above reduces boiler plate and makes for a fluent API

File: spec/rds/db_instance_spec.rb [hunk 6]
```diff
@@ -183,3 +296,14 @@
   its(:dbi_resource_id) { is_expected.to eq 'db-AAAAAAAAAAAAAAAAAAAAAAAAAA' }
   its(:ca_certificate_identifier) { is_expected.to eq 'rds-ca-2015' }
 end
+
+# rubocop:disable all
```
Comment (temyers) [inline spec/rds/db_instance_spec.rb]: rubocop needs to be re-enabled.

**Comment by temyers** [comment]
@SaltwaterC This PR is ready to go.  
I'll give you the opportunity to review before I merge.

**Comment by temyers** [comment]
@SaltwaterC Would you be able to push a release?

**Comment by SaltwaterC** [comment]
Sure. I'll need to tag the release and bump the version. I'll do it once I get to my office as I don't have the gem environment configured on this box.

**Comment by SaltwaterC** [comment]
Release pushed. Thank you for taking your contributions.

**Pull request merged**